x-pack/plugins/ui_actions_enhanced/public/drilldowns/url_drilldown/url_template.ts

```diff
@@ -5,130 +5,147 @@
+      const [str, searchString, valueString] = args.slice(0, -1) as [string, string, string];
+      if (typeof searchString !== 'string' || typeof valueString !== 'string')
+        throw new Error(
+          '[replace]: "searchString" and "valueString" parameters expected to be strings, but not a string or missing'
+        );
+      return String(str).split(searchString).join(valueString);
+    });
 
-handlebars.registerHelper('lowercase', (rawValue: unknown) => String(rawValue).toLowerCase());
-handlebars.registerHelper('uppercase', (rawValue: unknown) => String(rawValue).toUpperCase());
-handlebars.registerHelper('trim', (rawValue: unknown) => String(rawValue).trim());
-handlebars.registerHelper('trimLeft', (rawValue: unknown) => String(rawValue).trimLeft());
-handlebars.registerHelper('trimRight', (rawValue: unknown) => String(rawValue).trimRight());
-handlebars.registerHelper('left', (rawValue: unknown, numberOfChars: number) => {
-  if (typeof numberOfChars !== 'number')
-    throw new Error('[left]: expected "number of characters to extract" to be a number');
-  return String(rawValue).slice(0, numberOfChars);
-});
-handlebars.registerHelper('right', (rawValue: unknown, numberOfChars: number) => {
-  if (typeof numberOfChars !== 'number')
-    throw new Error('[left]: expected "number of characters to extract" to be a number');
-  return String(rawValue).slice(-numberOfChars);
-});
-handlebars.registerHelper('mid', (rawValue: unknown, start: number, length: number) => {
-  if (typeof start !== 'number') throw new Error('[left]: expected "start" to be a number');
-  if (typeof length !== 'number') throw new Error('[left]: expected "length" to be a number');
-  return String(rawValue).substr(start, length);
-});
-handlebars.registerHelper('concat', (...args) => {
-  const values = args.slice(0, -1) as unknown[];
-  return values.join('');
-});
-handlebars.registerHelper('split', (...args) => {
-  const [str, splitter] = args.slice(0, -1) as [string, string];
-  if (typeof splitter !== 'string') throw new Error('[split] "splitter" expected to be a string');
-  return String(str).split(splitter);
-});
-handlebars.registerHelper('replace', (...args) => {
-  const [str, searchString, valueString] = args.slice(0, -1) as [string, string, string];
-  if (typeof searchString !== 'string' || typeof valueString !== 'string')
-    throw new Error(
-      '[replace]: "searchString" and "valueString" parameters expected to be strings, but not a string or missing'
-    );
-  return String(str).split(searchString).join(valueString);
-});
+    handlebars.registerHelper('encodeURIComponent', (component: unknown) => {
+      const str = String(component);
+      return encodeURIComponent(str);
+    });
+    handlebars.registerHelper('encodeURIQuery', (component: unknown) => {
+      const str = String(component);
+      return url.encodeUriQuery(str);
+    });
+
+    return handlebars;
+  }));
+
+  function createSerializationHelper(
```

Comment:
nit: maybe would be nice to have this function in module scope instead of having it here inside another function.